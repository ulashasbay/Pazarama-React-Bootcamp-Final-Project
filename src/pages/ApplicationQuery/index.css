.app-check-container {
  font-family: "Open Sans", sans-serif;
  color: #3a3c47;
  line-height: 1.6;
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 0;
  padding: 0;
  height: auto;
}

.app-check-container h1 {
  margin-top: 48px;
  font-size: 30px;
}

.application-check-form {
  background: #fff;
  max-width: 360px;
  width: 100%;
  padding: 58px 44px;
  border: 1px solid #e1e2f0;
  border-radius: 4px;
  box-shadow: 0 0 5px 0 rgba(42, 45, 48, 0.12);
  transition: all 0.3s ease;
}

.app-check-row {
  display: flex;
  flex-direction: column;
  margin-bottom: 20px;
}

.app-check-row label {
  font-size: 13px;
  color: #8086a9;
}

.app-check-row input {
  flex: 1;
  padding: 13px;
  border: 1px solid #d6d8e6;
  border-radius: 4px;
  font-size: 16px;
  transition: all 0.2s ease-out;
}

.app-check-row input:focus {
  outline: none;
  box-shadow: inset 2px 2px 5px 0 rgba(42, 45, 48, 0.12);
}

.app-check-row input::placeholder {
  color: #c8cddf;
}

/* text area alanları için */
.app-check-row textarea {
  flex: 1;
  padding: 13px;
  border: 1px solid #d6d8e6;
  border-radius: 4px;
  font-size: 16px;
  transition: all 0.2s ease-out;
}

.app-check-row textarea:focus {
  outline: none;
  box-shadow: inset 2px 2px 5px 0 rgba(42, 45, 48, 0.12);
}

.app-check-row textarea::placeholder {
  color: #c8cddf;
}

.app-check-errors {
  margin-top: 0;
  font-size: 14px;
  color: red;
}

@media (max-width: 960px) {
  .app-check-container {
    margin: 0 18px;
  }

  .application-check-form {
    border: none;
    box-shadow: none;
    padding: 20px 0;
  }
}
