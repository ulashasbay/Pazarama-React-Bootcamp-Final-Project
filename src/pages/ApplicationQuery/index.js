import React from "react";
import { useFormik } from "formik";
import Input from "../../components/Input";
import FormSubmitBtn from "../../components/FormSubmitBtn";
import ApplicationQueryVal from "../../schema/ApplicationQueryVal";
import { doc, getDoc } from "firebase/firestore";
import { db } from "../../config/firebase";
import { useNavigate } from "react-router-dom";
import { useDispatch } from "react-redux";
import { updateAppId } from "../../redux/appId/appIdSlice";
import { updateUserInfo } from "../../redux/userInfo/userInfoSlice";
import "./index.css";

function ApplicationQuery() {
  let navigate = useNavigate();
  const dispatch = useDispatch();

  const getUserApplicationInfo = async (id) => {
    const noteSnapshot = await getDoc(doc(db, "applications", id));
    if (noteSnapshot.exists()) {
      dispatch(updateAppId(id));
      dispatch(updateUserInfo(noteSnapshot.data()));
      navigate(`/basvuru/${id}`);
    } else {
      navigate(`/404-Not-Found`);
      console.log("Böyle bir kullanıcı bulunmuyor");
    }
  };

  const { handleSubmit, handleChange, handleBlur, errors, touched } = useFormik(
    {
      initialValues: {
        applicationNo: "",
      },
      onSubmit: (values) => {
        getUserApplicationInfo(values.applicationNo);
      },
      validationSchema: ApplicationQueryVal,
    }
  );
  return (
    <div className="app-check-container">
      <h1>Başvuru Sorgula</h1>
      <form className="application-check-form" onSubmit={handleSubmit}>
        <div className="app-check-row">
          <Input
            name="applicationNo"
            type="text"
            text="Başvuru Numarası"
            placeholder="Başvuru No"
            onChange={handleChange}
            onBlur={handleBlur}
          />
          {errors.applicationNo && touched.applicationNo && (
            <span className="app-check-errors">{errors.applicationNo}</span>
          )}
        </div>
        <FormSubmitBtn text="Başvuru Sorgula" />
      </form>
    </div>
  );
}

export default ApplicationQuery;
