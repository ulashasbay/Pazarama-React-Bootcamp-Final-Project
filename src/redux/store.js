import { configureStore } from "@reduxjs/toolkit";
import appIdReducer from "./appId/appIdSlice";
import userInfoReducer from "./userInfo/userInfoSlice";
import isLoggedInReducer from "./isLoggedIn/isLoggedInSlice";

export const store = configureStore({
  reducer: {
    appId: appIdReducer,
    userInfo: userInfoReducer,
    isLoggedIn: isLoggedInReducer,
  },
});
