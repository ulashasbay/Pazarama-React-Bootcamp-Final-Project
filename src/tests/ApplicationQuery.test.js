import React from "react";
import { render, screen } from "@testing-library/react";
import ApplicationQuery from "../pages/ApplicationQuery";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import { store } from "../redux/store";
import renderer from "react-test-renderer";

const AllTheProviders = ({ children }) => {
  return (
    <BrowserRouter>
      <Provider store={store}>{children}</Provider>
    </BrowserRouter>
  );
};

describe("ApplicationQuery Tests", () => {
  let basvuruNoInput, button;
  beforeEach(() => {
    render(<ApplicationQuery />, { wrapper: AllTheProviders });
    basvuruNoInput = screen.getByPlaceholderText("Başvuru No");
    button = screen.getByRole("button");
  });

  test("Render Input", () => {
    expect(basvuruNoInput).toBeInTheDocument();
  });

  test("Başvuru No input type is text", () => {
    expect(basvuruNoInput.type).toEqual("text");
  });

  test("Render form label", () => {
    const basvuruNoLabel = screen.getByText("Başvuru Numarası");
    expect(basvuruNoLabel).toBeInTheDocument();
  });

  test("Render Başvuru Sorgula Button", () => {
    expect(button).toBeInTheDocument();
  });

  test("Başvuru Sorgula button type is submit", () => {
    expect(button.type).toEqual("submit");
  });
});

test("ApplicationQuery renders correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Provider store={store}>
          <ApplicationQuery />
        </Provider>
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
