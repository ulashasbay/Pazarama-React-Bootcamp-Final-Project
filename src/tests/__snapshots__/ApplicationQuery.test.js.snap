// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationQuery renders correctly 1`] = `
<div
  className="app-check-container"
>
  <h1>
    Başvuru Sorgula
  </h1>
  <form
    className="application-check-form"
    onSubmit={[Function]}
  >
    <div
      className="app-check-row"
    >
      <label
        htmlFor="applicationNo"
      >
        Başvuru Numarası
      </label>
      <input
        name="applicationNo"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Başvuru No"
        type="text"
      />
    </div>
    <button
      className="app-check-button"
      type="submit"
    >
      Başvuru Sorgula
    </button>
  </form>
</div>
`;
